stellargraph/data/edge_splitter.py

```diff
@@ -98,8 +98,10 @@ def _train_test_split_homogeneous(
         """
         # minedges are those edges that if removed we might end up with a disconnected graph after the positive edges
         # have been sampled.
-        if keep_connected:
+
+        if (not self.g.is_directed()) and keep_connected:
             self.minedges = self._get_minimum_spanning_edges()
+            self.minedges_set = set(self.minedges)
```

Comment:
```suggestion
```

It's slightly confusing, but `self.minedges_set` is set by `_get_minimum_spanning_edges`.

(I'm working on rewriting this, as part of #174.)

Reply:
ah I didn't realise